pkg/kubelet/kubelet_pods.go

```diff
@@ -1624,6 +1624,26 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		if err != nil {
 			klog.V(4).InfoS("Cannot get host IPs", "err", err)
 		} else {
+			if s.HostIP != "" {
+				ok, err := utilnet.IsDualStackIPStrings([]string{s.HostIP, hostIPs[0].String()})
+				if err != nil {
+					kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidHostIPs", "host IPs are invalid %q or %q", s.HostIP, hostIPs[0].String())
+				}
+				if ok {
+					// Once CloudDualStackNodeIPs is beta, <EXPLICIT-IPV4-ADDRESS>,<EXPLICIT-IPV6-ADDRESS> can be set to --node-ip.
+					if utilnet.IPFamilyOf(hostIPs[0]) == utilnet.IPv4 {
```

Comment:
if utilnet.IsIPv4(hostIPs[0]) {

actually, wait, this is backwards: `hostIPs` is what the cloud provider chose, `s.HostIP` is what kubelet detected.